src/Panel/Panel.php

```diff
@@ -45,7 +199,264 @@ public static function customCss(App $kirby)
     }
 
     /**
-     * Returns predefined icons path as sprite svg file
+     * Check for a custom js file from the
+     * config (panel.js)
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @return string|false
+     */
+    public static function customJs(App $kirby)
+    {
+        if ($js = $kirby->option('panel.js')) {
+            $asset = asset($js);
+
+            if ($asset->exists() === true) {
+                return $asset->url() . '?' . $asset->modified();
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Creates data array for Fiber and the component
+     *
+     * @param \Kirby\Cms\App $kirby
+     * @param array $data
+     * @return array
+     */
+    public static function data(App $kirby, array $data = []): array
+    {
+        // multilang setup check
+        $multilang = $kirby->option('languages', false) !== false;
+
+        // shared data for all requests
+        $shared = [
+            '$language' => function () use ($kirby, $multilang) {
+                if ($multilang === true && $language = $kirby->language()) {
+                    return [
+                        'code'    => $language->code(),
+                        'default' => $language->isDefault(),
+                        'name'    => $language->name(),
+                    ];
+                }
+            },
+            '$languages' => function () use ($kirby, $multilang): array {
+                if ($multilang === true) {
+                    return $kirby->languages()->values(function ($language) {
+                        return [
+                            'code'    => $language->code(),
+                            'default' => $language->isDefault(),
+                            'name'    => $language->name(),
+                        ];
+                    });
+                }
+
+                return [];
+            },
+            '$permissions' => function () use ($kirby) {
+                if ($user = $kirby->user()) {
+                    return $user->role()->permissions()->toArray();
+                }
+            },
+            '$license' => $kirby->system()->license(),
```

Comment:
Should this be passed with every response, even if the user is not admin?

Reply:
No, it shouldn't!! We should convert this to a boolean. The license key is only needed in the settings view and we can check for admin rights there before we send it.